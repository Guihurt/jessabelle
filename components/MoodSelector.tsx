import React from "react";
import { View, StyleSheet, TouchableOpacity } from "react-native";
import { FontAwesomeIcon } from "@fortawesome/react-native-fontawesome";
import {
  faMeh,
  faSmile,
  faGrinBeam,
  faFrown,
  faSadTear,
} from "@fortawesome/free-regular-svg-icons";

type MoodSelectorProps = {
  iconSize: number;
  setMood: Function;
  selectedMood: number;
};

export const Mood = {
  LOW: -2,
  MID_LOW: -1,
  MEDIUM: 0,
  MID_HIGH: 1,
  HIGH: 2,
} as const;

export default function MoodSelector({
  iconSize,
  setMood,
  selectedMood,
}: MoodSelectorProps) {
  const selectedStyle = (selectedMood: number, Mood: number) => {
    return selectedMood === Mood ? styles.selectedButton : {};
  };

  return (
    <View style={styles.container}>
      <TouchableOpacity
        style={[styles.moodButton, selectedStyle(selectedMood, Mood.LOW)]}
        onPress={() => setMood(Mood.LOW)}
      >
        <FontAwesomeIcon icon={faSadTear} size={iconSize} />
      </TouchableOpacity>
      <TouchableOpacity
        style={[styles.moodButton, selectedStyle(selectedMood, Mood.MID_LOW)]}
        onPress={() => setMood(Mood.MID_LOW)}
      >
        <FontAwesomeIcon icon={faFrown} size={iconSize} />
      </TouchableOpacity>
      <TouchableOpacity
        style={[styles.moodButton, selectedStyle(selectedMood, Mood.MEDIUM)]}
        onPress={() => setMood(Mood.MEDIUM)}
      >
        <FontAwesomeIcon icon={faMeh} size={iconSize} />
      </TouchableOpacity>
      <TouchableOpacity
        style={[styles.moodButton, selectedStyle(selectedMood, Mood.MID_HIGH)]}
        onPress={() => setMood(Mood.MID_HIGH)}
      >
        <FontAwesomeIcon icon={faSmile} size={iconSize} />
      </TouchableOpacity>
      <TouchableOpacity
        style={[styles.moodButton, selectedStyle(selectedMood, Mood.HIGH)]}
        onPress={() => setMood(Mood.HIGH)}
      >
        <FontAwesomeIcon icon={faGrinBeam} size={iconSize} />
      </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    backgroundColor: "white",
    justifyContent: "space-evenly",
    padding: 5,
    marginVertical: 5,
    borderRadius: 15,
  },
  moodButton: {
    paddingVertical: 2.5,
    paddingHorizontal: 2.5,
    flex: 1,
    alignItems: "center",
    borderRadius: 15,
  },
  selectedButton: {
    backgroundColor: "purple",
  },
});
