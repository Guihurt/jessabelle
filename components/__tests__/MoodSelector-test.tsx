import * as React from "react";
import renderer from "react-test-renderer";
import MoodSelector from "../MoodSelector/MoodSelector";

describe("<MoodSelector />", () => {
  const defaultSelector = renderer.create(<MoodSelector />).toJSON();
  it("renders, default props", () => {
    expect(defaultSelector).toMatchSnapshot();
  });
});
