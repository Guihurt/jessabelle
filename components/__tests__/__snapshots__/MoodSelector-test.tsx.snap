// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MoodSelector /> renders, default props 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 15,
      "flexDirection": "row",
      "justifyContent": "space-evenly",
      "marginVertical": 5,
      "padding": 5,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 15,
        "flex": 1,
        "opacity": 1,
        "paddingHorizontal": 2.5,
        "paddingVertical": 2.5,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={24}
      bbWidth={24}
      fill="#000"
      focusable={false}
      height={24}
      meetOrSlice={0}
      minX={0}
      minY={0}
      secondaryFill={null}
      secondaryOpacity={0.4}
      size={24}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {},
          Object {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      vbHeight={512}
      vbWidth={496}
      width={24}
    >
      <RNSVGGroup
        fill={4278190080}
        propList={
          Array [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 448c-110.3 0-200-89.7-200-200S137.7 56 248 56s200 89.7 200 200-89.7 200-200 200zm8-152c-13.2 0-24 10.8-24 24s10.8 24 24 24c23.8 0 46.3 10.5 61.6 28.8 8.1 9.8 23.2 11.9 33.8 3.1 10.2-8.5 11.6-23.6 3.1-33.8C330 320.8 294.1 304 256 304zm-88-64c17.7 0 32-14.3 32-32s-14.3-32-32-32-32 14.3-32 32 14.3 32 32 32zm160-64c-17.7 0-32 14.3-32 32s14.3 32 32 32 32-14.3 32-32-14.3-32-32-32zm-165.6 98.8C151 290.1 126 325.4 126 342.9c0 22.7 18.8 41.1 42 41.1s42-18.4 42-41.1c0-17.5-25-52.8-36.4-68.1-2.8-3.7-8.4-3.7-11.2 0z"
          fill={4278190080}
          fillOpacity={1}
          propList={
            Array [
              "fill",
              "fillOpacity",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 15,
        "flex": 1,
        "opacity": 1,
        "paddingHorizontal": 2.5,
        "paddingVertical": 2.5,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={24}
      bbWidth={24}
      fill="#000"
      focusable={false}
      height={24}
      meetOrSlice={0}
      minX={0}
      minY={0}
      secondaryFill={null}
      secondaryOpacity={0.4}
      size={24}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {},
          Object {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      vbHeight={512}
      vbWidth={496}
      width={24}
    >
      <RNSVGGroup
        fill={4278190080}
        propList={
          Array [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 448c-110.3 0-200-89.7-200-200S137.7 56 248 56s200 89.7 200 200-89.7 200-200 200zm-80-216c17.7 0 32-14.3 32-32s-14.3-32-32-32-32 14.3-32 32 14.3 32 32 32zm160-64c-17.7 0-32 14.3-32 32s14.3 32 32 32 32-14.3 32-32-14.3-32-32-32zm-80 128c-40.2 0-78 17.7-103.8 48.6-8.5 10.2-7.1 25.3 3.1 33.8 10.2 8.4 25.3 7.1 33.8-3.1 16.6-19.9 41-31.4 66.9-31.4s50.3 11.4 66.9 31.4c8.1 9.7 23.1 11.9 33.8 3.1 10.2-8.5 11.5-23.6 3.1-33.8C326 321.7 288.2 304 248 304z"
          fill={4278190080}
          fillOpacity={1}
          propList={
            Array [
              "fill",
              "fillOpacity",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "purple",
        "borderRadius": 15,
        "flex": 1,
        "opacity": 1,
        "paddingHorizontal": 2.5,
        "paddingVertical": 2.5,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={24}
      bbWidth={24}
      fill="#000"
      focusable={false}
      height={24}
      meetOrSlice={0}
      minX={0}
      minY={0}
      secondaryFill={null}
      secondaryOpacity={0.4}
      size={24}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {},
          Object {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      vbHeight={512}
      vbWidth={496}
      width={24}
    >
      <RNSVGGroup
        fill={4278190080}
        propList={
          Array [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 448c-110.3 0-200-89.7-200-200S137.7 56 248 56s200 89.7 200 200-89.7 200-200 200zm-80-216c17.7 0 32-14.3 32-32s-14.3-32-32-32-32 14.3-32 32 14.3 32 32 32zm160-64c-17.7 0-32 14.3-32 32s14.3 32 32 32 32-14.3 32-32-14.3-32-32-32zm8 144H160c-13.2 0-24 10.8-24 24s10.8 24 24 24h176c13.2 0 24-10.8 24-24s-10.8-24-24-24z"
          fill={4278190080}
          fillOpacity={1}
          propList={
            Array [
              "fill",
              "fillOpacity",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 15,
        "flex": 1,
        "opacity": 1,
        "paddingHorizontal": 2.5,
        "paddingVertical": 2.5,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={24}
      bbWidth={24}
      fill="#000"
      focusable={false}
      height={24}
      meetOrSlice={0}
      minX={0}
      minY={0}
      secondaryFill={null}
      secondaryOpacity={0.4}
      size={24}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {},
          Object {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      vbHeight={512}
      vbWidth={496}
      width={24}
    >
      <RNSVGGroup
        fill={4278190080}
        propList={
          Array [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 448c-110.3 0-200-89.7-200-200S137.7 56 248 56s200 89.7 200 200-89.7 200-200 200zm-80-216c17.7 0 32-14.3 32-32s-14.3-32-32-32-32 14.3-32 32 14.3 32 32 32zm160 0c17.7 0 32-14.3 32-32s-14.3-32-32-32-32 14.3-32 32 14.3 32 32 32zm4 72.6c-20.8 25-51.5 39.4-84 39.4s-63.2-14.3-84-39.4c-8.5-10.2-23.7-11.5-33.8-3.1-10.2 8.5-11.5 23.6-3.1 33.8 30 36 74.1 56.6 120.9 56.6s90.9-20.6 120.9-56.6c8.5-10.2 7.1-25.3-3.1-33.8-10.1-8.4-25.3-7.1-33.8 3.1z"
          fill={4278190080}
          fillOpacity={1}
          propList={
            Array [
              "fill",
              "fillOpacity",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 15,
        "flex": 1,
        "opacity": 1,
        "paddingHorizontal": 2.5,
        "paddingVertical": 2.5,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={24}
      bbWidth={24}
      fill="#000"
      focusable={false}
      height={24}
      meetOrSlice={0}
      minX={0}
      minY={0}
      secondaryFill={null}
      secondaryOpacity={0.4}
      size={24}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {},
          Object {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      vbHeight={512}
      vbWidth={496}
      width={24}
    >
      <RNSVGGroup
        fill={4278190080}
        propList={
          Array [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 448c-110.3 0-200-89.7-200-200S137.7 56 248 56s200 89.7 200 200-89.7 200-200 200zm105.6-151.4c-25.9 8.3-64.4 13.1-105.6 13.1s-79.6-4.8-105.6-13.1c-9.8-3.1-19.4 5.3-17.7 15.3 7.9 47.1 71.3 80 123.3 80s115.3-32.9 123.3-80c1.6-9.8-7.7-18.4-17.7-15.3zm-235.9-72.9c3.5 1.1 7.4-.5 9.3-3.7l9.5-17c7.7-13.7 19.2-21.6 31.5-21.6s23.8 7.9 31.5 21.6l9.5 17c2.1 3.7 6.2 4.7 9.3 3.7 3.6-1.1 6-4.5 5.7-8.3-3.3-42.1-32.2-71.4-56-71.4s-52.7 29.3-56 71.4c-.3 3.7 2.1 7.2 5.7 8.3zm160 0c3.5 1.1 7.4-.5 9.3-3.7l9.5-17c7.7-13.7 19.2-21.6 31.5-21.6s23.8 7.9 31.5 21.6l9.5 17c2.1 3.7 6.2 4.7 9.3 3.7 3.6-1.1 6-4.5 5.7-8.3-3.3-42.1-32.2-71.4-56-71.4s-52.7 29.3-56 71.4c-.3 3.7 2.1 7.2 5.7 8.3z"
          fill={4278190080}
          fillOpacity={1}
          propList={
            Array [
              "fill",
              "fillOpacity",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
